inference-engine/src/inference_engine/include/openvino/runtime/executable_network.hpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file that provides ExecutableNetwork class
+ *
+ * @file ie_executable_network.hpp
+ */
+
+#pragma once
+
+#include <map>
+#include <memory>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "ie_parameter.hpp"
+#include "infer_request.hpp"
+#include "remote_context.hpp"
+
+namespace InferenceEngine {
+class IExecutableNetworkInternal;
+class RemoteContext;
+}  // namespace InferenceEngine
+namespace ov {
+class Function;
+namespace op {
+namespace v0 {
+class Parameter;
+class Result;
+}  // namespace v0
+}  // namespace op
```

Comment:
I think we need to include `#include <openvino/core/function.hpp` directly

Reply:
@ilyachur how do you think?

Reply:
In nGraph we don't use the forward declaration and include necessary headers. I propose to use the unified code approach.

Reply:
done